rust/arrow/src/array/array_binary.rs

```diff
@@ -373,6 +376,45 @@ impl From<Vec<Vec<u8>>> for FixedSizeBinaryArray {
     }
 }
 
+impl From<Vec<Option<Vec<u8>>>> for FixedSizeBinaryArray {
+    fn from(data: Vec<Option<Vec<u8>>>) -> Self {
+        let len = data.len();
+        assert!(len > 0);
+        // try to estimate the size. This may not be possible no entry is valid => panic
+        let size = data.iter().filter_map(|e| e.as_ref()).next().unwrap().len();
+        assert!(data
```

Comment:
Given this operation can fail (if all the elements are not the same length) perhaps we should implement `TryFrom` instead of `From` and `panic` -- again, this would be an excellent follow on PR (or maybe file a ticket and it could be an excellent first contribution for someone who wanted to contribute)